eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -0,0 +1,192 @@
+<#
+.SYNOPSIS
+Update unified ToC file for publishing reference docs on docs.microsoft.com
+
+.DESCRIPTION
+Given a doc repo location and a location to output the ToC generate a Unified
+Table of Contents:
+
+* Get list of packages onboarded to docs.microsoft.com (domain specific)
+* Get metadata for onboarded packages from metadata CSV
+* Build a sorted list of services
+* Add ToC nodes for the service
+* Add "Core" packages to the bottom of the ToC under "Other"
+
+ToC node layout:
+* Service (service level overview page)
+  * Client Package 1 (package level overview page)
+  * Client Package 2 (package level overview page)
+  ...
+  * Management
+    * Management Package 1
+    * Management Package 2
+    ...
+
+.PARAMETER DocRepoLocation
+Location of the documentation repo. This repo may be sparsely checked out
+depending on the requirements for the domain
+
+.PARAMETER OutputLocation
+Output location for unified reference yml file
+
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [string] $DocRepoLocation,
+
+  [Parameter(Mandatory = $true)]
+  [string] $OutputLocation
+)
+. $PSScriptRoot/common.ps1
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+function GetClientPackageNode($clientPackage) {
+  $packageInfo = &$GetDocsMsTocData `
+    -packageMetadata $clientPackage `
+    -docRepoLocation $DocRepoLocation
+
+  return [PSCustomObject]@{
+    name     = $packageInfo.PackageTocHeader
+    href     = $packageInfo.PackageLevelReadmeHref
+    # This is always one package and it must be an array
+    children = $packageInfo.TocChildren
+  };
+}
+
+$onboardedPackages = &$GetOnboardedDocsMsPackages `
+  -DocRepoLocation $DocRepoLocation
+
+# This criteria is different from criteria used in `Update-DocsMsPackages.ps1`
+# because we need to generate ToCs for packages which are not necessarily "New"
+# in the metadata AND onboard legacy packages (which `Update-DocsMsPackages.ps1`
+# does not do)
+$metadata = (Get-CSVMetadata).Where({
+    $_.Package `
+      -and $onboardedPackages.ContainsKey($_.Package) `
+      -and $_.Hide -ne 'true'
+  })
+
+# Map metadata to package. Validate metadata entries.
+$packagesForToc = @{}
+foreach ($packageName in $onboardedPackages.Keys) {
```

Comment:
Not sure how or if it is worth it sharing this but if we do need the same thing for other languages we should add a helper. I also have a similar helper in the azure-sdk repo https://github.com/Azure/azure-sdk/blob/main/eng/scripts/PackageList-Helpers.ps1#L178.

Reply:
This looks like it can work. It might make sense to put it in eng/common/scripts though its usage is only scoped to `Update-DocsMsToc.ps` for now.

Reply:
Yeah something under eng-common that builds a look-up table once we have multiple users.